### PR TITLE
Add Canonicalize option to Locale Constraint

### PR DESCRIPTION
**Pull request by zairigimad**
Add Canonicalize option to Locale Constraint (which was added in Symfony 4.1: https://symfony.com/index.php/blog/new-in-symfony-4-1-validator-improvements)

### CODE REVIEW

File: reference/constraints/Locale.rst [hunk 1]
```diff
@@ -111,6 +112,12 @@ Parameter        Description
 
 .. include:: /reference/constraints/_payload-option.rst.inc
 
+``canonicalize``
+~~~~~~~~~~~
```
Comment (OskarStark) [inline reference/constraints/Locale.rst]:
```suggestion
~~~~~~~~~~~~~~~~
```

File: reference/constraints/Locale.rst [hunk 2]
```diff
@@ -111,6 +112,12 @@ Parameter        Description
...
+**type**: ``boolean`` **default**: ``true``
```
Comment (OskarStark) [inline reference/constraints/Locale.rst]:
```suggestion

**type**: ``boolean`` **default**: ``true``
```

File: reference/constraints/Locale.rst [hunk 3]
```diff
@@ -111,6 +112,12 @@ Parameter        Description
...
+
+this method performs Level 1 and Level 2 canonicalization according to ICU standards `ICU format locale IDs`.
```
Comment (OskarStark) [inline reference/constraints/Locale.rst]:
```suggestion
this method performs level 1 and level 2 canonicalization according to ICU standards `ICU format locale IDs`.
```

**Comment by OskarStark** [comment]
@ro0NL can you please have a look here? Thanks!

File: reference/constraints/Locale.rst (hunk 3 above)
Comment (ro0NL) [reply reference/constraints/Locale.rst]: uppercase `This ...`

**Review by ro0NL — COMMENTED** [review]
:+1: https://www.php.net/manual/en/locale.canonicalize.php#118376

**Comment by OskarStark** [comment]
For 3.4 I guess?

**Review by OskarStark — APPROVED** [review]

File: reference/constraints/Locale.rst [hunk 4]
```diff
@@ -111,6 +112,13 @@ Parameter        Description
 
 .. include:: /reference/constraints/_payload-option.rst.inc
 
+``canonicalize``
+~~~~~~~~~~~~~~~~
+
+**type**: ``boolean`` **default**: ``true``
+
+This method performs level 1 and level 2 canonicalization according to ICU standards `ICU format locale IDs`.
```
Comment (OskarStark) [inline reference/constraints/Locale.rst]:
```suggestion
This method performs level 1 and level 2 canonicalization according to
ICU standards `ICU format locale IDs`.
```

File: reference/constraints/Locale.rst [hunk 5]
```diff
@@ -17,6 +17,7 @@ Applies to  :ref:`property or method <validation-property-target>`
 Options     - `groups`_
             - `message`_
             - `payload`_
+            _ `canonicalize`_
```
Comment (OskarStark) [inline reference/constraints/Locale.rst]: @javiereguiluz shall we sort them in alpha order?

**Comment by javiereguiluz** [comment]
I remembered that this option was already documented. It was merged here: https://github.com/symfony/symfony-docs/pull/9248/files . So I don't understand how/when it was lost 🤔

**Comment by OskarStark** [comment]
@javiereguiluz are you able to finalize this PR?

**Comment by wouterj** [comment]
This option was introduced in 4.1 by https://github.com/symfony/symfony/pull/26075 as a backwards compatibility layer. In 4.1 setting it to anything different than `true` was deprecated by https://github.com/symfony/symfony/pull/26075 and this setting was removed in Symfony 5.0 (keeping a setting that is only allowed to be `true` is non-sense) in https://github.com/symfony/symfony/pull/31898

We then correctly also removed this option from the Symfony 5.0+ docs in
https://github.com/symfony/symfony-docs/commit/ce64ac294bf9ec5479fa0b9a1976b5b24d7433b4#diff-08df604c59d5284048fac6277839ca0dL92

To me, it seems perfectly fine that this option is not documented in 5.0 and master. In the 4.4 version of the docs, this option is still visible: https://symfony.com/doc/4.4/reference/constraints/Locale#canonicalize 

For these reasons, I'm closing this PR.

**Comment by ro0NL** [comment]
@wouterj where did we remove this in 5.x?

https://github.com/symfony/symfony/blob/332fa65f69d2cf929b9f3f07a64c2bd236995c58/src/Symfony/Component/Validator/Constraints/Locale.php#L33

:thinking: 

AFAIK only the default changed in between versions.

**Pull request closed without merging**